Automattic/wp-calypso
# Fix signup with-theme flow for Marketplace themes and other improvements

Fix the with-theme flow for Marketplace themes and filter the plans based on the type of the theme:
- For Marketplace and DotOrg themes we will now show just `Business` and `eCommerce` plans.
- For Premium themes, we will now show `Premium`, `Business` and `eCommerce` plans - this is needed because the user will be redirected to designSetup step where they'll be asked yet again to purchase `Premium`.

In both cases, we still display the `start with free` link in the header. I think we need to allow users an escape hatch, in case they don't want to purchase a qualified plan for the theme.

### TODO in a separate PR
You'll notice that there are some discrepancies here for sites that select an incompatible plan (e.g. free for premium themes): 
- Marketplace themes, Users are redirected to DesignSetup theme list where they can select a different theme
- DotORG themes, users are redirected to /marketplace/install/ where an upgrade nudge is displayed asking for an upgrade to Business
- Premium themes, users are redirected to a dedicated /DesignSetup page for a given theme, where they are asked again to upgrade.

We should come up with a flow consistent for all flows, as much as possible, given the current limitations (e.g. dot-org and Marketplace themes don't work on DesignSetup).



Related to https://github.com/Automattic/wp-calypso/issues/75335

## Proposed Changes

* Add a new plans step that handles the with-theme flows, especially the Marketplace themes.

## Testing Instructions



* Go incognito to WordPress.com/themes
* Right-click on the Pick design button and copy the /start/with-theme link
* Open a browser window where you are already logged-in and go to that URL.
* Make sure that for Marketplace themes:
* * we only display the Business and eCommerce plans
* * After selecting one of the plans, you are redirected to checkout with the plan added and with the theme subscription selected
* * Make sure that when you select a yearly/monthly plan subscription, the theme subscription has the same recurrence
* * Make sure that after you are purchasing the theme and plan, you are now getting an AT site with the theme installed.
* * Make sure that when you click on the `or Start with free`, you are redirected to the DesignSetup theme list, where you can select another theme
* Make sure that for Premium themes we only display Premium, Business and eCommerce plans
* Make sure that for DotORG themes we only display Business and eCommerce plans
## Pre-merge Checklist
